### PR TITLE
fix: Update Snowflake spec and docs

### PR DESCRIPTION
**Pull request by yevgenypats**
I realised the `dsn` in snowflake spec is inconsistent with other plugins so changed it to `connection_string` and added a recipe + docs. Blog will come after that in a separate PR.

### CODE REVIEW

File: website/pages/docs/recipes/destinations/snowflake.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+Note: Make sure you use environment variable expansion in production instead of committing the credentials to the configuration file directly.
+
+## Basic
+
+```yaml copy
+kind: destination
+spec:
+  name: snowflake
+  path: cloudquery/snow
```
Comment (hermanschaaf) [inline website/pages/docs/recipes/destinations/snowflake.md:13]:
```suggestion
  path: cloudquery/snowflake
```
?

File: website/pages/docs/recipes/destinations/snowflake.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  version: "v1.0.1" # latest version of postgresql plugin
```
Comment (hermanschaaf) [inline website/pages/docs/recipes/destinations/snowflake.md:14]:
```suggestion
  version: "v1.0.1" # latest version of snowflake plugin
```

File: website/pages/docs/recipes/destinations/snowflake.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  spec:
+    connection_string: ${SNOW_CONNECTION_STRING}
```
Comment (hermanschaaf) [inline website/pages/docs/recipes/destinations/snowflake.md:16]:
```suggestion
    connection_string: ${SNOWFLAKE_CONNECTION_STRING}
```

File: website/pages/docs/recipes/destinations/snowflake.md (hunk 2 above)
Comment (yevgenypats) [reply website/pages/docs/recipes/destinations/snowflake.md:14]: Good catch. My upcoming PR should solve variables over strings hopefully :)

**Review by hermanschaaf — APPROVED** [review]

**Pull request merged**